galaxy_ng/app/api/v3/serializers/sync.py

```diff
@@ -127,6 +144,31 @@ class Meta:
             'ca_cert': {'write_only': True},
         }
 
+    def to_representation(self, instance):
+
+        if instance.proxy_url is not None:
+            address, username, _ = strip_auth_from_url(instance.proxy_url)
+            instance.proxy_address = address
+            instance.proxy_username = username
+            # instance.proxy_password is write_only, no need to represent.
+            
+        return super().to_representation(instance)
+
+    def save(self):
+
+        proxy_address = self.validated_data.get('proxy_address')
+        proxy_username = self.validated_data.get('proxy_username')
+        proxy_password = self.validated_data.get('proxy_password')
+
+        if proxy_address is not None:
+            self.validated_data['proxy_url'] = join_proxy_url(
+                proxy_address,
+                proxy_username,
+                proxy_password,  # TODO: Or default to existing password from proxy_url on DB.
```

Comment:
When user changes username but does't provide a password
the password is going to be set to None

to fix that we have 2 options:

- require password to be set whenever username is changed
- default password to the existing password on database

but this required us to do a database lookup here and have UI to manage the empty state (when user wants to clean up the password)

@newswangerd

Reply:
If `proxy_password` is missing from the api request then proxy password should be the same as the existing value in the DB. If `proxy_password` is set to `null` or blank, the password should be removed.